sql/update.go

```diff
@@ -18,11 +18,99 @@
 package sql
 
 import (
+	"fmt"
+
+	"github.com/cockroachdb/cockroach/client"
 	"github.com/cockroachdb/cockroach/sql/parser"
-	"github.com/cockroachdb/cockroach/util"
+	"github.com/cockroachdb/cockroach/util/log"
 )
 
-// Update TODO(pmattis): document.
+// Update updates columns for a selection of rows from a table.
+// TODO(vivek): Update columns that are a part of an index.
 func (p *planner) Update(n *parser.Update) (planNode, error) {
-	return nil, util.Errorf("TODO(pmattis): unimplemented")
+	tableDesc, err := p.getAliasedTableDesc(n.Table)
+	if err != nil {
+		return nil, err
+	}
+
+	// Determine which columns we're inserting into.
+	var names parser.QualifiedNames
+	for _, expr := range n.Exprs {
+		names = append(names, expr.Name)
+	}
+	cols, err := p.processColumns(tableDesc, names)
```

Comment:
`UPDATE` is not allowed to change "primary key" columns, but I don't see anything here preventing doing so. You should add a test case and any necessary code here to prevent such updates.